zellij-server/src/screen.rs

```diff
@@ -188,22 +188,40 @@ impl Screen {
         }
     }
 
+    /// A helper function to switch to a new tab at specified position.
+    fn switch_active_tab(&mut self, new_tab_pos: usize) {
+        let current_tab = self.get_active_tab().unwrap();
+        let current_tab_pos = current_tab.position;
+
+        // If new active tab is same as the current one, do nothing.
+        if current_tab_pos == new_tab_pos {
+            return;
+        }
+
+        // FIXME: old active tab should only be inactive once the new active tab is determined,
```

Comment:
@TheLostLambda Do you have any idea about this ？

Reply:
Took a while, but what do you think about this? Does this still work?

```rs
    /// A helper function to switch to a new tab at specified position.
    fn switch_active_tab(&mut self, new_tab_pos: usize) {
        if let Some(new_tab_idx) = self.tabs.values().position(|t| t.position == new_tab_pos) {
            let current_tab = self.get_active_tab().unwrap();

            // If new active tab is same as the current one, do nothing.
            if current_tab.position == new_tab_idx {
                return;
            }

            current_tab.inactive();

            let new_tab = self.get_indexed_tab_mut(new_tab_idx).unwrap();
            new_tab.set_force_render();
            new_tab.active();

            let old_active_index = self.active_tab_index.replace(new_tab_idx);
            self.tab_history.retain(|&e| e != Some(new_tab_idx));
            self.tab_history.push(old_active_index);

            self.update_tabs();
            self.render();
        }
    }
```

Reply:
Wow, looks very nice! Thanks a lot!

Reply:
@TheLostLambda 
Sorry, I'm happy too fast.

This is also forbidden:

```
warning: cannot borrow `*self` as mutable because it is also borrowed as immutable
   --> zellij-server/src/screen.rs:202:27
    |
194 |         if let Some(new_tab) = self.tabs.values().find(|t| t.position == new_tab_pos) {
    |                                --------- immutable borrow occurs here
...
202 |             let new_tab = self.get_indexed_tab_mut(new_tab.index).unwrap();
    |                           ^^^^                     ------------- immutable borrow later used here
    |                           |
    |                           mutable borrow occurs here
    |
    = note: `#[warn(mutable_borrow_reservation_conflict)]` on by default
    = warning: this borrowing pattern was not meant to be accepted, and may become a hard error in the future
    = note: for more information, see issue #59159 <https://github.com/rust-lang/rust/issues/59159>

error[E0499]: cannot borrow `self.active_tab_index` as mutable more than once at a time
   --> zellij-server/src/screen.rs:206:36
    |
202 |             let new_tab = self.get_indexed_tab_mut(new_tab.index).unwrap();
    |                           ---- first mutable borrow occurs here
...
206 |             let old_active_index = self.active_tab_index.replace(new_tab.index);
    |                                    ^^^^^^^^^^^^^^^^^^^^^         ------------- first borrow later used here
    |                                    |
    |                                    second mutable borrow occurs here
```